src/rgw/driver/sfs/object.cc

```diff
@@ -23,15 +23,23 @@ using namespace std;
 namespace rgw::sal {
 
 SFSObject::SFSReadOp::SFSReadOp(SFSObject *_source)
-    : source(_source) {}
+    : source(_source) {
+  /*
+    This initialization code was originally into prepare() but that
+    was not sufficient to cover all cases.
+    There are pieces of SAL code that are calling get_*() methods
+    but they don't call prepare().
+    In those cases the SFSReadOp is not properly initialized and those
+    calls are going to fail.
+  */
+  source->refresh_meta();
```

Comment:
Any side effects to this? refresh_meta makes a database access and may throw

Reply:
are you suggesting to `try..catch()` every occurrence of `refresh_meta` in `sfs` code?

Reply:
Not suggesting that. Asking what the side effects are to moving that to the constructor. Possible errors will now occur there not in prepare. Is that a problem?

Reply:
unfortunately hard to say; I have no idea if that would have been a problem even in the previous location.